src/codewhisperer/util/dependencyGraph/rubyDependencyGraph.ts

```diff
@@ -0,0 +1,235 @@
+import * as CodeWhispererConstants from '../../models/constants'
+import { readFileAsString } from '../../../shared/filesystemUtilities'
+import { sleep } from '../../../shared/utilities/timeoutUtils'
+import { DependencyGraphConstants, DependencyGraph, Truncation } from './dependencyGraph'
+import path = require('path')
+
+export const importRegex = /(require|require_relative|load|include|extend)\s+('[^']+'|"[^"]+"|\w+)(\s+as\s+(\w+))?/gm
+
+export class RubyDependencyGraph extends DependencyGraph {
+    // Payload Size for Ruby: 1MB
+    getPayloadSizeLimitInBytes(): number {
+        return CodeWhispererConstants.codeScanRubyPayloadSizeLimitBytes
+    }
+    override updateSysPaths(uri: vscode.Uri) {
+        this.getDirPaths(uri).forEach(dirPath => {
+            this._sysPaths.add(dirPath)
+        })
+    }
+
+    private generateFilePath(modulePath: string, dirPath: string) {
+        const filePath = path.join(dirPath, modulePath + DependencyGraphConstants.rubyExt)
+        return existsSync(filePath) ? filePath : ''
+    }
+
+    //For Generating File Paths
+    private generateFilePaths(modulePaths: string[], dirPaths: string[]) {
+        return modulePaths
+            .flatMap(modulePath => dirPaths.map(dirPath => this.generateFilePath(modulePath, dirPath)))
+            .filter(filePath => filePath !== '')
+    }
+
+    //Generate the combinations for module paths
+    private generateModulePaths(inputPath: string): string[] {
+        const positionOfExt = inputPath.indexOf(DependencyGraphConstants.rubyExt) //To remove imports having .rb
+        if (positionOfExt !== -1) {
+            inputPath = inputPath.substring(0, positionOfExt).trim()
+        }
+
+        const inputPaths = inputPath.split('/')
+        let outputPath = ''
+        return inputPaths.map(pathSegment => {
+            outputPath += (outputPath ? '/' : '') + pathSegment
+            return path.join(...outputPath.split('/'))
+        })
+    }
+
+    private getModulePath(modulePathStr: string) {
+        const pos = modulePathStr.indexOf(DependencyGraphConstants.as)
+        if (pos !== -1) {
+            modulePathStr = modulePathStr.substring(0, pos)
+        }
+
+        return this.generateModulePaths(modulePathStr.replace(/[",'\s()]/g, '').trim())
+    }
+
+    private extractModulePaths(importStr: string) {
+        let modulePaths: string[] = []
+        const {
+            require: requireKeyword,
+            require_relative: requireRelativeKeyword,
+            include: includeKeyword,
+            extend: extendKeyword,
+            load: loadKeyword,
+        } = DependencyGraphConstants
+
+        let keyword: string | undefined
+        let indexOfKeyword: number | undefined
+
+        switch (true) {
+            case importStr.startsWith(requireRelativeKeyword):
+                keyword = requireRelativeKeyword
+                indexOfKeyword = importStr.indexOf(requireRelativeKeyword)
+                break
```

Comment:
Seems like checking the `indexOf` keyword is unnecessary because `startsWith` is already used

Reply:
Updated accordingly!